airflow/providers/google/cloud/hooks/pubsub.py

```diff
@@ -144,25 +146,27 @@ def _validate_messages(messages) -> None:
                 try:
                     b64decode(message["data"])
                     warnings.warn(
+                        "During message validation, the 'data' field was successfully decoded as base64. "
```

Comment:
Just because the input can be decoded as a base64-encoded string doesn't mean it is a base64-encoded string. I took the liberty of adding a clarifying message to the warning.